chalicelib/checks/helpers/lifecycle_utils.py

```diff
@@ -0,0 +1,299 @@
+LONG_TERM_ACCESS_LONG_TERM_ARCHIVE = "long_term_access_long_term_archive"
+LONG_TERM_ACCESS = "long_term_access"
+SHORT_TERM_ACCESS = "short_term_access"
+LONG_TERM_ARCHIVE = "long_term_archive"
+SHORT_TERM_ARCHIVE = "short_term_archive"
+NO_STORAGE = "no_storage"
+IGNORE = "ignore"
+
+
+MOVE_TO_INFREQUENT_ACCESS_AFTER = "move_to_infrequent_access_after"
+MOVE_TO_GLACIER_AFTER = "move_to_glacier_after"
+MOVE_TO_DEEP_ARCHIVE_AFTER = "move_to_deep_archive_after"
+EXPIRE_AFTER = "expire_after"
+STANDARD = "standard"
+INFREQUENT_ACCESS = "infrequent access"
+GLACIER = "glacier"
+DEEP_ARCHIVE = "deep archive"
+DELETED = "deleted"
+PENDING = "pending"
+COMPLETE = "complete"
+UPLOADED = "uploaded"
+ARCHIVED = "archived"
+
+
+default_lifecycle_policy = {
+    SHORT_TERM_ACCESS_LONG_TERM_ARCHIVE: {
+        MOVE_TO_INFREQUENT_ACCESS_AFTER: 0,
+        MOVE_TO_DEEP_ARCHIVE_AFTER: 3,
+        EXPIRE_AFTER: 36,
+    },
+    LONG_TERM_ACCESS_LONG_TERM_ARCHIVE: {
+        MOVE_TO_INFREQUENT_ACCESS_AFTER: 0,
+        MOVE_TO_DEEP_ARCHIVE_AFTER: 12,
+        EXPIRE_AFTER: 36,
+    },
+    SHORT_TERM_ACCESS: {
+        MOVE_TO_INFREQUENT_ACCESS_AFTER: 0,
+        EXPIRE_AFTER: 12,
+    },
+    LONG_TERM_ACCESS: {
+        MOVE_TO_INFREQUENT_ACCESS_AFTER: 0,
+        EXPIRE_AFTER: 36,
+    },
+    SHORT_TERM_ARCHIVE: {
+        MOVE_TO_DEEP_ARCHIVE_AFTER: 0,
+        EXPIRE_AFTER: 12,
+    },
+    LONG_TERM_ARCHIVE: {
+        MOVE_TO_DEEP_ARCHIVE_AFTER: 0,
+        EXPIRE_AFTER: 36,
+    },
+    NO_STORAGE: {
+        EXPIRE_AFTER: 0,
+    },
+}
+
+lifecycle_status_to_file_status = {
+    STANDARD: UPLOADED,
+    INFREQUENT_ACCESS: UPLOADED,
+    GLACIER: ARCHIVED,
+    DEEP_ARCHIVE: ARCHIVED,
+    DELETED: DELETED
+}
+
+
+def check_file_lifecycle_status(num_files_to_check, first_check_after, max_checking_frequency, my_auth):
+    """
+    This main check function. Factored out for easier testing
+    """
+
+    check_result = {
+        "status": "PASS",
+        "warning": ""
+    }
+
+    # We only want to get files from the portal that have a lifecycle category set and have either never been checked
+    # or previously checked sufficiently long ago - as far as I know this can't be combined into one query. Furthermore,
+    # they should be at least {first_check_after} days old
+    threshold_date_fca = datetime.date.today() - datetime.timedelta(first_check_after)
+    threshold_date_fca = threshold_date_fca.strftime("%Y-%m-%d")
+    threshold_date_mcf = datetime.date.today() - datetime.timedelta(max_checking_frequency)
+    threshold_date_mcf = threshold_date_mcf.strftime("%Y-%m-%d")
+
+    search_query_base = (
+        "/search/?type=FileProcessed&type=FileFastq"
+        "&s3_lifecycle_category%21=No+value"
+        f"&s3_lifecycle_category%21={IGNORE}"
+        f"&date_created.to={threshold_date_fca}"
+        "&status=uploaded"
+        "&status=archived"
+        "&status=shared"
```

Comment:
Should we broaden this to take into account additional file statuses? Why wouldn't we want to include those with a status of `in review` or `current`? It may be easier to define the status we don't want to search on here.

Reply:
As discussed, we are using `"&status%21=deleted"`